peer/peer.go

```diff
@@ -1820,6 +1813,37 @@ func (p *Peer) readRemoteVersionMsg() error {
 	return nil
 }
 
+// addrmgrToWireNetAddress converts an IPv4, IPv6, or TORv2 address manager
+// network address to a wire network address.  The host name must not match the
+// configured proxy host name.
+//
+// If the address cannot be converted, an IPv4 network address consisting of all
+// zeroes is returned.
+func addrmgrToWireNetAddress(addr *addrmgr.NetAddress, proxyAddr string) *wire.NetAddress {
```

Comment:
I'm a little bit worried about how this is being split out and named as if it is for generic use, but it's not due to the additional proxy handling.  It's really specifically for use during version negotiation.